### ISSUE TITLE
Unit toxiproxy.service not found

### ISSUE THREAD

**Issue by bronzdoc** [Shopify/toxiproxy#205]
Hi :wave: 

I'm trying to install toxiproxy following the ubuntu instructions: 
```
$ wget -O toxiproxy-2.1.3.deb https://github.com/Shopify/toxiproxy/releases/download/v2.1.3/toxiproxy_2.1.3_amd64.deb
$ sudo dpkg -i toxiproxy-2.1.3.deb
$ sudo service toxiproxy start
```

But i'm getting `Failed to start toxiproxy.service: Unit toxiproxy.service not found.` after running `$ sudo service toxiproxy start`

**Comment by jpittis** [comment]
Thanks for opening an issue @bronzdoc. I'll get to this as soon as I can.

**Comment by zackkitzmiller** [comment]
Any update on this? I'm running into the same problem.

**Comment by jpittis** [comment]
Sorry for such a huge delay on this issue @zackkitzmiller. I just booted up a fresh Ubuntu install and successfully reproduced this.

You'll notice that after installing the package, you should have access to `toxiproxy-server` which you'll be able to run from your command line to synchronously start Toxiproxy. Unfortunately, this will block your console... I'm looking into why the daemon isn't starting now.

**Comment by jpittis** [comment]
I'm not super comfortable with all this daemon stuff but I'll try to help debug.

Alright so I see `/etc/init/toxiproxy.conf` on my machine so the conf file is present as it should be.

So `file /sbin/init` shows me `/sbin/init: symbolic link to /lib/systemd/systemd` which I think means my machine is running systemd.

So my understanding is that `/etc/init` is for Upstart and `/etc/init.d` is for systemd. And because we're both running a systemd machine, it doesn't work with Toxiproxy's Upstart config.

**So tldr: You're on a systemd machine and toxiproxy only supports Upstart. IMO we should support both systemd as well as upstart. I'm going to tag this issue as needing a fix.**

**Comment by jpittis** [comment]
You're correct that this issue has yet to be fixed.

> toxiproxy can't run on systemd

It can run, you just have to configure it manually to do so.

It still works as per usual when executing it directly.

**Comment by jimkleine** [comment]
@jpittis, I'm new here, so I haven't yet worked out how to submit a pull request for a package file, however I can assist with a solution.

Upstart is deprecated on Ubuntu after 16.04 and there is no Upstart package available for 18.04.  

I have created a `toxiproxy.service` using `/etc/init/toxiproxy.conf` as a template.  I am testing with 18.04.  The `toxiproxy.service` file needs to be created in `/lib/systemd/system`. The service can then be started with `sudo systemctl start toxiproxy`.

```
[Unit]
Description=TCP proxy to simulate network and system conditions
After=network-online.target firewalld.service
Wants=network-online.target

[Service]
Type=simple
Environment=HOST=localhost
Environment=PORT=8474
ExecStart=/usr/bin/toxiproxy-server -port $PORT -host $HOST
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

**Comment by jpittis** [comment]
Hey @jimkleine! Appreciate you putting this together.

- [Here's our current toxiproxy.conf file.](https://github.com/Shopify/toxiproxy/blob/master/share/toxiproxy.conf)
- [Here's how it's hooked into our release process.](https://github.com/Shopify/toxiproxy/blob/master/Makefile#L67)

Given this information, I'm not sure how to proceed. Hopefully  I'll have some time next week to think it through and get back to you with more info. Unless you know the obvious solution to this problem, in which case go ahead a submit a PR!

**Comment by jimkleine** [comment]
@jpittis Thanks for hinting me in the right direction (the link to the `Makefile` was the missing piece).  Just give me a couple of days to integrate GitHub into my workflow and polish my GitHub etiquette and I should be able to create a pull request.

NB This project has a failing build tag at the moment.  That looks like a result of switching from CircleCI to Travis: the project is still reporting the CircleCI status.

**Comment by jimkleine** [comment]
@jpittis FYI: The build badge is part of the markdown code in README.md.  I'll create a separate pull request with just that change.  I realise that is technically a separate issue, so I won't discuss it further here.

**Comment by OnkelTem** [comment]
Omg. You could just have included the file provided by @jimkleine until you have a better solution because now it is not working out of the box for 100% of Ubuntu users.

**Comment by trance1st** [comment]
I can confirm that the file provided by @jimkleine works. Please use it to fix this issue

**Comment by Ladicek** [comment]
Also confirmed that the unit file provided by @jimkleine works. I submitted a PR (#296), where I tried to tag @jimkleine as a co-author (hope that's fine with you @jimkleine!). I'll have to ask internally whether I can sign the CLA, though.

**Comment by arne-da** [comment]
Note: even though this issue was closed, the issue still persists since #296 wasn't merged.

**Comment by miry** [comment]
@arne-da We drop supporting linux distribution. Contribution are welcome. Here is more information how it could be possibly done: https://github.com/Shopify/toxiproxy/pull/296#issuecomment-907067232

**Comment by Kle0s** [comment]
I think that since it's required as a part of the *Getting Started* on Ubuntu (I personally use 20.04), it should be at least mentioned somewhere, as it might throw people off

**Comment by bennycode** [comment]
I just encountered the same error:

> Failed to start toxiproxy.service: Unit toxiproxy.service not found.

This happened after following the installation steps from the README:

``` 
wget -O toxiproxy-2.1.4.deb https://github.com/Shopify/toxiproxy/releases/download/v2.1.4/toxiproxy_2.1.4_amd64.deb
sudo dpkg -i toxiproxy-2.1.4.deb
sudo service toxiproxy start
``` 

My system is running Ubuntu 24.04.1 LTS.

Is there any guidance on how to fix this?

**Comment by jimkleine** [comment]
Hi @bennycode , have you tried manually creating the file per the sample I offered above?  Note that I haven't specifically tested it with 24.04, but I can if it doesn't work for you.

I'll try to make time to create a permanent fix, but need to review the history first.

> I just encountered the same error:
> 
> > Failed to start toxiproxy.service: Unit toxiproxy.service not found.
> 
> This happened after following the installation steps from the README:
> 
> ``` 
> wget -O toxiproxy-2.1.4.deb https://github.com/Shopify/toxiproxy/releases/download/v2.1.4/toxiproxy_2.1.4_amd64.deb
> sudo dpkg -i toxiproxy-2.1.4.deb
> sudo service toxiproxy start
> ``` 
> 
> My system is running Ubuntu 24.04.1 LTS.
> 
> Is there any guidance on how to fix this?
